dnf/cli/cli.py

```diff
@@ -972,6 +972,24 @@ def configure(self, args, option_parser=None):
         if self.base.conf.color != 'auto':
             self.base.output.term.reinit(color=self.base.conf.color)
 
+        if rpm.expandMacro('%_pkgverify_level') in ('signature', 'all'):
+            forcing = False
+            for repo in self.base.repos.iter_enabled():
+                if repo.gpgcheck:
+                    continue
+                repo.gpgcheck = True
```

Comment:
A note/nit/reminder: We should really introduce a `pkg_gpgcheck` option for managing this, and introduce some kind of `gpgcheck_policy` flag for setting the default behavior of `gpgcheck` (whether it governs both pkg+repo or just pkg or just repo). The naming and current behavior is really dumb and confusing.

Reply:
We already have a distinction between the pkg/repo checking: `gpgcheck` and `repo_gpgcheck`. But yeah, we also have `localpkg_gpgcheck` which makes it a bit awkward. I agree there should ideally be one `gpgcheck` option with a bunch of possible values/policies (similar to how the `pkgverify_level` macro handles that, for example).

I'm hoping we'll get around to cleaning up the whole gpgcheck story in libdnf in some not too distant future, and when we do, I'll definitely push for making things simple and easy to understand.

Reply:
The funny thing is that there's some ten year old comment in the yum codebase to do exactly what I mention... And in fact `zypper` already does it this way for this reason.